### DISCUSSION TITLE
ScopedElementsMixin

### DISCUSSION THREAD

**Discussion by JakobVogelsang** [open-wc/open-wc#2824] [Q&A] [answered]
Hi all, I am working on a application and are using web components a lot. I would like to allow others to use my components and want to make those side effect free. As some of my components are a composition of other of my web components I though I give the scoped custom registry a try. It does not work for me in all scenarios and I am wondering why. Did I missunderstand the capability or do I do something wrong. This is one of the scenarious that I cannot make work.

Component1.ts
```ts
import { html, LitElement } from "lit"

export class MyComponent1 extends LitElement {
    render() {
        return html`<input value="Component1"/><button>Component1</button>`
    }
}
```

Component2.ts
```ts
import { html, LitElement } from "lit"

import { ScopedElementsMixin } from "@open-wc/scoped-elements/html-element.js";

import { MyComponent1 } from "./Component1.js";

export class MyComponent2 extends ScopedElementsMixin(LitElement) {
    static scopedElements = {
        'my-component-1': MyComponent1,
      };

    render() {
        return html`<div>This is compoent 2 rendering component 1</div>
        <my-component-1></my-component-1>`
    }
}
```

Component3.ts
```ts
import { LitElement, html } from 'lit';

import { ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';

import { MyComponent1 } from './Component1.js';
import { MyComponent2 } from './Component2.js';

export class MyComponent3 extends ScopedElementsMixin(LitElement) {
  static scopedElements = {
    'my-component-2': MyComponent2,
    'my-component-1': MyComponent1,
  };

  render() {
    return html`This is the 1st component <my-component-1></my-component-1>
                        <my-component-2></my-component-2>`;
  }
}
```

my-component-3.ts
```ts
import { MyComponent3 } from "./Component3.js";

customElements.define('my-component-3', MyComponent3);
```

That loaded into a simepe index.html lead to the result as such:
<img width="497" alt="grafik" src="https://github.com/user-attachments/assets/7a236099-91e6-4bd7-b9b2-a0fb57041c0c" />

Wehre the DOM looks like so:
<img width="533" alt="grafik" src="https://github.com/user-attachments/assets/220d8f01-7561-44fa-8c7c-9566b741c593" />

Why is `my-component-1` loaded with `my-compoent-3` correctly but not within `my-component-2`????

**Comment by thepassle** [comment, marked as answer]
In component2 you're importing:
```js
import { ScopedElementsMixin } from "@open-wc/scoped-elements/html-element.js";
```

Can you try importing from:
```js
import { ScopedElementsMixin } from "@open-wc/scoped-elements/lit-element.js";
```
instead?

**Reply by JakobVogelsang** [reply]
Ohhh for f*** **** **** . How could I not see this!! Thx
